src/PHPStanStaticTypeMapper/Utils/TypeUnwrapper.php

```diff
@@ -39,6 +39,10 @@ public function removeNullTypeFromUnionType(UnionType $unionType): UnionType
             $unionedTypesWithoutNullType[] = $type;
         }
 
+        if ($unionedTypesWithoutNullType !== []) {
```

Comment:
on else, union with empty types is impossible, I think this can be handled by `count($unionedTypesWithoutNullType) === 1`

Reply:
https://github.com/rectorphp/rector-src/pull/6220